packages/date-time-picker/test/value-commit.test.js

```diff
@@ -0,0 +1,376 @@
+      await expectNoValueCommit();
+    });
+
+    it('should commit as unparsable on date-time-picker outside click', async () => {
+      outsideClick();
+      await expectUnparsableValueCommit();
+    });
+  });
+
+  describeForEachPicker('unparsable input entered', (pickerType) => {
+    let picker, otherPicker;
+
+    beforeEach(async () => {
+      [picker, otherPicker] = getPicker(pickerType);
+      await enterUnparsableInput(picker);
+      resetSpyHistories();
+    });
+
+    it('should commit as unparsable on picker Enter', async () => {
+      await sendKeys({ press: 'Enter' });
+      await expectUnparsableValueCommit();
+    });
+
+    it('should commit as unparsable on other picker click', async () => {
+      await sendMouseToElement({ type: 'click', element: otherPicker });
+      await expectUnparsableValueCommit();
+    });
+
+    it('should commit as unparsable on date-time-picker on outside click', async () => {
+      outsideClick();
+      await expectUnparsableValueCommit();
+    });
+  });
+
+  describeForEachPicker('complete input entered', (pickerType) => {
+    let picker, otherPicker;
+
+    beforeEach(async () => {
+      [picker, otherPicker] = getPicker(pickerType);
+      await enterParsableInput(otherPicker);
+      await sendKeys({ press: 'Enter' });
+      await enterParsableInput(picker);
+      resetSpyHistories();
+    });
+
+    it('should commit on picker Enter', async () => {
+      await sendKeys({ press: 'Enter' });
+      await nextRender();
+      await expectValueCommit();
+    });
+
+    it('should commit on other picker click', async () => {
+      await sendMouseToElement({ type: 'click', element: otherPicker });
+      await expectValueCommit();
+    });
+
+    it('should commit on date-time-picker outside click', async () => {
+      outsideClick();
+      await expectValueCommit();
+    });
+  });
+
+  describe('initially valid value', () => {
+    beforeEach(async () => {
+      dateTimePicker.value = '2002-02-02T12:00';
+      await nextRender();
+      resetSpyHistories();
+    });
+
+    describeForEachPicker('value partially cleared (= incomplete value)', (pickerType) => {
+      let picker, otherPicker;
+
+      beforeEach(async () => {
+        [picker, otherPicker] = getPicker(pickerType);
+        await clearInput(picker);
+      });
+
+      it('should not commit on Enter', async () => {
+        await sendKeys({ press: 'Enter' });
+        await expectNoValueCommit();
+      });
+
+      it('should not commit on other picker click', async () => {
+        await clearInput(picker);
+        await sendMouseToElement({ type: 'click', element: otherPicker });
+        await expectNoValueCommit();
+      });
+
+      it('should commit on date-time-picker outside click', async () => {
+        await clearInput(picker);
+        outsideClick();
+        await expectValueCommit();
+      });
+    });
+  });
+  describe('initially invalid value', () => {
```

Comment:
Let's also add tests for an initially invalid empty field, which can take place when a required datetimepicker is blurred without any input. This is the opposite scenario to `initially invalid value`.

```js
describe('initially invalid empty field', () => {
  beforeEach(() => {
    dateTimePicker.required = true;
    dateTimePicker.validate();
  });

  describeForEachPicker('incomplete input entered', () => {
     // ...
  });
})
```